### DISCUSSION TITLE
best easy way to add appendix

### DISCUSSION THREAD

**Discussion by majdgh92** [typst/typst#4031] [General]
Hello everyone,
i have some text to add as appendix in my work.
i would like to have an an appendix section after the content. I would like to have the name (Anlagen) for it.
So after list of photos and tables there should be a section for list of appendixes, which should also be added to content.
could someone please provide me with a detailed way to do this.
Thanks in advance.

**Comment by Mc-Zen** [comment]
Hi, I've also wondered about that. I think, a very neat and typsty solution would be to have this in your template:
```typ
#let appendix(body) = {
  set heading(numbering: "A", supplement: [Appendix])
  body
}
```
(or similar). When you feel the need for it, you can now call
```typ
#show: appendix
```
in your document and from now on, headings will be shown in the new style (regarding numbering). As a nice side-effect, you'll get `Appendix A` etc. if you reference your appendices by label - just as one would expect. 

Now let's move on to the outline. With the work we've done beforehand, this is actually surprisingly easy, because the outline `target` lets you use selectors:
```typ
#outline(target: heading.where(supplement: [Appendix]), title: [Appendix])
```

Let me know if this solved your quest.

**Reply by majdgh92** [reply]
Thank you very much, the only problem is that it starts from letter D.
= my title
it will show as D my title.

**Reply by Mc-Zen** [reply]
Ah, right. I missed this one because I had no numbering activated in the "main part" :D
Just add `counter(heading).update(0)` to the appendix show rule. In total like this: 

```typ

#let appendix(body) = {
  set heading(numbering: "A", supplement: [Appendix])
  counter(heading).update(0)
  body
}
#set heading(numbering: "1")
= Introduction
= Conclusion
For more info, see @app1. 

#show: appendix

= Tables and Data <app1>
= Additional Listings <app2>

#outline(target: heading.where(supplement: [Appendix]), title: [Appendix])
```
![image](https://github.com/typst/typst/assets/52877387/a867956b-e934-44d1-b502-0f03e930ca76)

**Reply by majdgh92** [reply]
Thank you very much. it works now as i wanted.

**Reply by Mc-Zen** [reply]
You're welcome.

**Reply by majdgh92** [reply]
one last question please, how to link to specific appendix in text?

**Reply by Mc-Zen** [reply]
Hi, I already added that to the example above. Just put a label on the heading and reference it as usual.

**Comment by JakobTopholt** [comment]
Really a nice discussion here, I just can't quite seem to get it to work. Currently I have a body starting from 1.Introduction.typ to X.nameOfFile.typ in a folder named 1Sections, and a folder named 2Appendix with a file 1.nameOfFile.typ and I want all files from that folder to be appendix entries starting from A.

I am using the [IEEE template ](https://github.com/bsp0109/ieee-typst-template/blob/main/template.typ), where every .typ file imports the template, it doesn't work just simply inserting
```
#let appendix(body) = {
  set heading(numbering: "A", supplement: [Appendix])
  counter(heading).update(0)
  body
}
```

and inserting #show: appendix in the appendix file(s) (there is currently only one)

As I do so I am informed that appendix is an unknown variable where i  #show: appendix and in main.typ there  i use #include "2Appendix/1.nameOfFile.typ"

I was somewhat hoping to rubber duck my way out of this by writing it out, but alas it did not work

**Reply by Mc-Zen** [reply]
Hi, I would suggest you call `show:appendix` from `main.typ` before including the other files. Unlike with latex, code declared imported by the root file (here  main.typ) is not available in the included files in Typst afaik!

Does that help?

**Comment by LevitatingBusinessMan** [comment]
I am using `charged-ieee:0.1.3` and I cannot get the heading to update.

**Reply by hiimnhan** [reply]
Do you have any solution yet?

**Reply by LevitatingBusinessMan** [reply]
I used this function:

```typst
/// altered version of https://github.com/typst/templates/blob/main/charged-ieee/lib.typ
/// DOES NOT WORK WITH DEEPER LEVELS
#let appendix(body) = {
  set heading(numbering: "A.1", supplement: [Appendix])
  counter(heading).update(0)
  show heading: it => {
    set text(11pt, weight: 400)
    set align(center)
    show: block.with(above: 15pt, below: 13.75pt, sticky: true)
    show: smallcaps
    [#it.supplement #counter(heading).display():]
    h(0.3em)
    it.body
  }
  body
}
```

And then used `#show: appendix` before the appendix.

**Reply by hiimnhan** [reply]
it worked! Thank you so much

**Reply by richli** [reply]
The IEEE style[^1] for appendices uses this:

```
Appendix A
Appendix Name
```

Rather than this:

```
Appendix A: Appendix Name
```

So I used the above `appendix` function with this diff:

```diff
     show: smallcaps
-    [#it.supplement #counter(heading).display():]
-    h(0.3em)
+    [#it.supplement #counter(heading).display()]
+    "\n"
     it.body
```

And it worked, thanks!

[^1]: See [the IEEE editorial style manual for authors](https://journals.ieeeauthorcenter.ieee.org/your-role-in-article-production/ieee-editorial-style-manual/), under "Text Section Headings"
